mtp_api/apps/notification/management/commands/send_notification_report.py

```diff
@@ -0,0 +1,297 @@
+    elif period_end > period_start + datetime.timedelta(days=7):
+        raise CommandError('Maximum date span is 7 days')
+
+    return period_start, period_end
+
+
+def generate_report(report_path, rule, record_set, serialiser):
+    with report_path.open('w') as f:
+        writer = csv.writer(f)
+        writer.writerow(serialiser.get_headers())
+        count = 0
+        for record in record_set:
+            if rule.applies_to(record) and rule.triggered(record):
+                writer.writerow(serialiser.serialise(record))
+                count += 1
+        return count
+
+
+def create_zip(period_filename, bundle_path):
+    report_paths = bundle_path.glob('*.csv')
+    report_paths = filter(lambda p: p.stat().st_size > 0, report_paths)
+    report_paths = sorted(report_paths)
+    if not report_paths:
+        return None
+
+    zip_path = bundle_path / f'{period_filename}.zip'
+    with zipfile.ZipFile(zip_path, 'w') as z:
+        for path in report_paths:
+            z.write(path, path.name)
+    return zip_path
+
+
+def send_zip(period_description, zip_file, emails):
+    email = AnymailMessage(
+        subject=f'Prisoner money notifications for {period_description}',
+        body=f'''
+OFFICIAL SENSITIVE
+
+Please find attached, the prisoner money notifications reports for {period_description}.
+
+If you have any queries, contact {settings.TEAM_EMAIL}.
+        '''.strip(),
+        from_email=default_from_address(),
+        to=emails,
+        tags=['notifications-report'],
+    )
+    email.attach_file(str(zip_file), mimetype='application/zip')
+    email.send()
+
+
+class Serialiser:
+    def __init__(self, rule):
+        self.rule = rule
+        self.monitored_rule = isinstance(rule, MonitoredRule)
+
+    def get_headers(self):
+        headers = [
+            'Notification rule',
+            'Internal ID',
+        ]
+        if self.monitored_rule:
+            headers.insert(1, 'Monitored by')
+        return headers
+
+    def serialise(self, record):
+        row = [
+            self.rule.description.replace('you are monitoring', 'someone monitors'),
+            self.internal_id(record),
+        ]
+        if self.monitored_rule:
+            row.insert(1, self.rule.get_event_trigger(record).get_monitoring_users().count())
+        return row
+
+    def internal_id(self, record):
+        raise NotImplementedError
+
+
+class CreditSerialiser(Serialiser):
+    def get_headers(self):
+        return super().get_headers() + [
+            'Date received', 'Date credited',
+            'Amount',
+            'Prisoner number', 'Prisoner name', 'Prison',
+            'Sender name', 'Payment method',
+            'Bank transfer sort code', 'Bank transfer account', 'Bank transfer roll number',
+            'Debit card number', 'Debit card expiry', 'Debit card billing address',
+            'Email', 'IP address',
+            'Status',
+            'NOMIS transaction',
```

Comment:
One value per line would be easier to read I think, same for the `serialise` method